subscriber/simple_supply_subscriber/decoding.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+from simple_supply_addressing.addresser import AddressSpace
+from simple_supply_addressing.addresser import get_address_type
+from simple_supply_protobuf.agent_pb2 import AgentContainer
+from simple_supply_protobuf.record_pb2 import RecordContainer
+
+
+CONTAINERS = {
+    AddressSpace.AGENT: AgentContainer,
+    AddressSpace.RECORD: RecordContainer
+}
+
+
+def data_to_dicts(address, data):
+    data_type = get_address_type(address)
+
+    if data_type == AddressSpace.OTHER_FAMILY:
+        return []
+
+    try:
+        container = CONTAINERS[data_type]
+    except KeyError:
+        raise TypeError('Unknown data type: {}'.format(data_type))
+
+    entries = _parse_proto(container, data).entries
+    return data_type, [_proto_to_dict(pb) for pb in entries]
+
+
+def _parse_proto(proto_class, data):
+    deserialized = proto_class()
+    deserialized.ParseFromString(data)
+    return deserialized
+
+
+def _proto_to_dict(proto):
+    result = {}
+
+    for field in proto.DESCRIPTOR.fields:
+        key = field.name
+        value = getattr(proto, key)
+
+        if field.type == field.TYPE_MESSAGE:
+            if field.label == field.LABEL_REPEATED:
+                result[key] = [_proto_to_dict(p) for p in value]
```

Comment:
How deep is this recursion expected to go?

Reply:
The Simple Supply protos only have one level of repeated fields, though this method should be able to handle any level necessary

Reply:
Apologies for commenting here, I am getting `maximum recursion depth exceeded in comparison` when this code executes. Know any workaround?